command/v7/create_service_key_command.go

```diff
@@ -0,0 +1,108 @@
+package v7
+
+import (
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/actor/v7action"
+	"code.cloudfoundry.org/cli/command/flag"
+	"code.cloudfoundry.org/cli/command/v7/shared"
+	"code.cloudfoundry.org/cli/types"
+)
+
+type CreateServiceKeyCommand struct {
+	BaseCommand
+
+	RequiredArgs     flag.ServiceInstanceKey       `positional-args:"yes"`
+	ParametersAsJSON flag.JSONOrFileWithValidation `short:"c" description:"Valid JSON object containing service-specific configuration parameters, provided either in-line or in a file. For a list of supported configuration parameters, see documentation for the particular service offering."`
+	Wait             bool                          `short:"w" long:"wait" description:"Wait for the operation to complete"`
+	relatedCommands  interface{}                   `related_commands:"service-key"`
+}
+
+func (cmd CreateServiceKeyCommand) Execute(args []string) error {
+	if err := cmd.SharedActor.CheckTarget(true, true); err != nil {
+		return err
+	}
+
+	if err := cmd.displayIntro(); err != nil {
+		return err
+	}
+
+	stream, warnings, err := cmd.Actor.CreateServiceKey(v7action.CreateServiceKeyParams{
+		SpaceGUID:           cmd.Config.TargetedSpace().GUID,
+		ServiceInstanceName: cmd.RequiredArgs.ServiceInstance,
+		ServiceKeyName:      cmd.RequiredArgs.ServiceKey,
+		Parameters:          types.OptionalObject(cmd.ParametersAsJSON),
+	})
+	cmd.UI.DisplayWarnings(warnings)
+	switch err.(type) {
+	case nil:
+	case actionerror.ResourceAlreadyExistsError:
+		cmd.displayAlreadyExists()
+		return nil
+	default:
+		return err
+	}
+
+	completed, err := shared.WaitForResult(stream, cmd.UI, cmd.Wait)
+	switch {
+	case err != nil:
+		return err
+	case completed:
+		cmd.UI.DisplayOK()
+		return nil
+	default:
+		cmd.UI.DisplayOK()
+		cmd.UI.DisplayText("Create in progress.")
+		return nil
+	}
+}
+
+func (cmd CreateServiceKeyCommand) Usage() string {
+	return `
+CF_NAME create-service-key SERVICE_INSTANCE SERVICE_KEY [-c PARAMETERS_AS_JSON] [--wait]
```

Comment:
We should change PARAMETERS_AS_JSON to JSON_PARAMETERS as per notes in story

Reply:
This triggered an investigation into what all the commands do to ensure consistency. Almost all commands use `PARAMETERS_AS_JSON` so we decided to standardise on that in this commit: https://github.com/cloudfoundry/cli/commit/146fa478c2aec6cdf93ab8833959d8cac3127b38